src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/usecases/env/EnvVariableTest.kt

```diff
@@ -0,0 +1,79 @@
+package it.krzeminski.snakeyaml.engine.kmp.usecases.env
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldEndWith
+import it.krzeminski.snakeyaml.engine.kmp.api.Load
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+import it.krzeminski.snakeyaml.engine.kmp.env.EnvConfig
+import it.krzeminski.snakeyaml.engine.kmp.exceptions.MissingEnvironmentVariableException
+import it.krzeminski.snakeyaml.engine.kmp.stringFromResources
+
+class EnvVariableTest : FunSpec({
+    // the variable is set by Gradle
+    val VALUE1 = "EnvironmentValue1"
```

Comment:
TODO: check the support for reading env vars for targets other than the JVM, currently the tests fail for these.

Reply:
Kotest has a method to read env variables. Probably, we could copy that part from them:

https://github.com/kotest/kotest/blob/master/kotest-common/src/commonMain/kotlin/io/kotest/mpp/sysprop.kt

Reply:
There is also a library that implements that already:

https://klibs.io/project/luca992/getenv-kt

Reply:
FTR, for JS we now have https://github.com/krzema12/snakeyaml-engine-kmp/blob/85a31ae35427bee23ec2801718d3143b55eaa4ac/src/jsCommonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/internal/system.js.kt#L3-L5
which explains why it doesn't work despite setting the vars in Gradle, and for native, there's a chance it will work:
https://github.com/krzema12/snakeyaml-engine-kmp/blob/85a31ae35427bee23ec2801718d3143b55eaa4ac/src/nativeMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/internal/system.native.kt#L10-L12

Reply:
The impl of https://klibs.io/project/luca992/getenv-kt is so simple I'll probably just copy it over to our lib, thanks for finding it!

Reply:
It is simple for all platforms except for `wasmWasi`. If/when this platform is supported by SnakeKMP we probably should use the `getenv-kt` instead

Reply:
Hmm, right. I'd like to avoid adding an extra dependency for now (various consequences), but you're right that in the future we may want to revisit.